myoFlex/myoFlex.ino

```diff
@@ -1,69 +1,131 @@
+  Serial.println("linking servos");
+  thumb.attach(thumbPin, min, max);    //defines where the servos are and their max/mins
+  pointer.attach(pointerPin, min, max);
+  middle.attach(middlePin, min, max);
+  ring.attach(ringPin, min, max);
+  pinky.attach(pinkyPin, min, max);
+  Serial.println("Starting");       //makes the hand relax all of the way
+  handPosition(max,max,max,max,max);
+  trigger = setTrigger();
+}
+
+//setup starts by linking the Servo objects to each pin and detailing the limits per each argument. Then it opens the hand for a default position and will call the setTrigger function. Once it's collected, the setup is complete
+
+void loop() {
+  Serial.println(analogRead(myoIn));
+  if(check(currentVoltage = analogRead(myoIn))){//collects voltage, then assigns to currentVoltage, then passes currentVoltage to check function, then checks the check return value to see if it's higher than trigger value
+    state++; //increases our state
+    state = state%4; //ensoures our only values are between 0-3 and loops back if otherwise
+    Serial.print("state is ");
+    Serial.println(state);
+    //Serial.println(currentVoltage);        //prints in the serial moniter
+    if(state == 0){
+      Serial.println("Open");
+      handPosition(max,max,max,max,max);
+    }
+    else if(state == 1){
+      Serial.println("Point");
+      handPosition(max,min,max,max,max);
+    }
+    else if(state == 2){
+      Serial.println("Pinch");
+      handPosition(min,min,min,max,max);
+    }
+    else if(state == 3){
+      Serial.println("Fist");
+      handPosition(min,min,min,min,min);
+    }
+    else {
+      state = 1; //edgecase if logic fail
+      handPosition(max,max,max,max,max);
+    }
+  }
+}
+
+//the loop function will continously check the users input and assign it to currentVoltage, then it's verified through the check function to see if its higher than the set trigger value from setup, if so then it will switch between the four hand positions in a looping structure
+
+int setTrigger(){
+  float average = 0;
+  int i, j;
+  do{
+    average = 0;//resets collection value if it failed prior
+    for(j = 1; j <= 3; j++){//loops to collect the three impulses
+      Serial.print("Reading impulse ");
+      Serial.print(j);
+      Serial.println(" in the next 3 seconds");
+      for(i = 3; i != 0; i--){//counts down 3 seconds
+        Serial.println(i);
+        delay(1000);
+      }
+      Serial.println("Reading");
+      average = average + analogRead(myoIn);//adds up collected values with previous values
+      Serial.println(average/3);
+    }
+  } while((average/3) < 200);//if it's less than 200 than we reattempt collection
+  return (int)(average/3);//returns average upon success
+}
+
+//setTrigger collects 3 inputs by the user with a 3 second delay between each one. It calculates the average and if it's lower than 200, it's an assumed error and will reset the collection. Once completed it returns the average
```

Comment:
Chris,

I like this function, but from my understanding it simply adds all of the readings over three seconds. I see two issues with this:

1) Instead of dividing by three, we should instead divide by the number of reads the myo gets for a true average.
2) I think we should instead look for the average _peak magnitude, not just the average signal. I think this will be easier to collect data on, as it is only three peaks to average, and also will be easier to scale an all or none response.

We should talk more later about this as it is a very interesting thing--it involves the question of do we want a toggle or continuous grip, but i think either of those should be based on the maximum contraction strength. 

Let me know what you think, and I can rewrite this section if you desire.

JD